plugin/metricstore/prometheus/metricstore/reader.go

```diff
@@ -60,16 +61,32 @@ type (
 		metricDesc        string
 		buildPromQuery    func(p promQueryParams) string
 	}
+
+	promClient struct {
+		api.Client
+		params map[string]string
+	}
 )
 
-// NewMetricsReader returns a new MetricsReader.
-func NewMetricsReader(cfg config.Configuration, logger *zap.Logger, tracer trace.TracerProvider) (*MetricsReader, error) {
-	logger.Info("Creating metrics reader", zap.Any("configuration", cfg))
+// URL decorator enables adding additional query parameters to the request sent to prometheus backend
+func (p promClient) URL(ep string, args map[string]string) *url.URL {
+	url := p.Client.URL(ep, args)
 
```

Comment:
I was originally thinking that passing union of `args` and `p.params` should work here, but it didn't. When I looked the implementation of `URL` here, I found this:

```
func (c *httpClient) URL(ep string, args map[string]string) *url.URL {
	p := path.Join(c.endpoint.Path, ep)

	for arg, val := range args {
		arg = ":" + arg
		p = strings.ReplaceAll(p, arg, val)
	}

	u := *c.endpoint
	u.Path = p

	return &u
}
```

I can't deduce what they are trying to do here, nor can I find any explanation. @yurishkuro can you help me with it?

Reply:
Oh nvm. I checked this: https://github.com/prometheus/client_golang/blob/main/api/client_test.go . It is not related to query parameters.

Reply:
it looks like it's doing placeholder replacement where placeholders are in the form `:name`.